Trying to find the variance between BDE values and the features/descriptors given by MORDRED

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn import datasets, linear_model
from sklearn.metrics import mean_squared_error, r2_score

In [2]:
features_df = pd.read_csv('SolventMordredResults_batch0.csv', on_bad_lines='skip')
features_df.head(5)


/var/folders/qc/kdqppv890tsc3zsxkk7rypjm0000gs/T/ipykernel_64804/3590084035.py:1: DtypeWarning: Columns (4,5,6,7,8,9,10,11,12,13,14,15,53,54,55,56,57,58,59,60,61,135,136,137,138,139,140,141,142,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,162,163,164,165,166,167,168,169,171,172,173,174,175,176,177,178,180,181,182,183,184,185,186,187,189,190,191,192,193,194,195,196,198,199,200,201,202,203,204,205,207,208,209,210,211,212,213,214,216,217,218,219,220,221,222,223,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,260,261,262,263,264,265,266,267,268,341,342,343,344,345,346,347,348,349,351,352,353,354,355,356,357,358,360,361,362,363,364,365,366,367,368,369,370,371,372,373,374,375,376,378,379,380,381,382,383,384,385,387,388,389,390,391,392,393,394,396,397,398,399,400,401,402,403,405,406,407,408,409,410,411,412,414,415,416,417,418,419,420,421,423,424,425,426,427,428,429,430,432,433,434,435,436,437,438,439,441,442,443,444,445,446,447,448,449,450,451,452,45

ABC     ABCGG  nAcid  nBase   SpAbs_A   SpMax_A  SpDiam_A    SpAD_A  \
0  4.949747  5.143137      0      0  9.517541  1.879385   3.75877  9.517541   
1  5.059137  4.785854      0      0  8.720566  2.101003  4.202006  8.720566   
2  1.414214  1.414214      0      0  2.828427  1.414214  2.828427  2.828427   
3  4.242641  4.000000      0      0       8.0       2.0       4.0       8.0   
4  3.047207  3.305183      0      0  5.226252  1.847759  3.695518  5.226252   

    SpMAD_A   LogEE_A  ...     SRW10     TSRW10          MW       AMW WPath  \
0  1.189693  2.876615  ...  7.126891  32.187603  114.140851  4.390033    84   
1  1.245795  2.844305  ...  8.124151  33.544698   92.062600  6.137507    42   
2  0.942809  1.849457  ...  4.174387  17.310771   46.041865  5.115763     4   
3  1.333333  2.687624  ...  7.627057  30.941317   88.052429  6.289459    27   
4   1.04525  2.408576  ...  6.834109  27.254130   72.057515  5.542886    18   

  WPol  Zagreb1  Zagreb2           mZagreb1  mZagreb2  
0    5     26.0     24.0                3.5  2.250000  
1    5     30.0     31.0  2.361111111111111  1.666667  
2    0      6.0      4.0               2.25  1.000000  
3    3     24.0     24.0                1.5  1.500000  
4    2     16.0     14.0  3.361111111111111  1.333333  

[5 rows x 1613 columns]

So, there are 1613 different features/descriptors in the dataset. We need to figure out which are independent

In [3]:
SMILES_df = pd.read_csv('SoluteSmiles.txt', on_bad_lines='skip')
SMILES_df.drop_duplicates()


Empty DataFrame
Columns: [CCCCCCCC,  Cc1ccccc1,  CCO,  C1COCCO1,  CCC(C)=O,  Cc1ccccn1,  c1ccc2ccccc2c1,  Oc1ccccc1,  Cc1cccc(O)c1,  CC(O)=O,  CCC(O)=O,  CCCC(O)=O,  CN(C)C,  CCNCC,  c1ccncc1,  Nc1ccccc1,  N,  CN,  CC(C)CC(C)=O,  CCCCCCCC,  Cc1ccccc1.1,  CCO.1,  C1COCCO1.1,  CCC(C)=O.1,  CC(O)=O.1,  C[N+]([O-])=O,  CCCCCCCC.1,  Cc1ccccc1.2,  CCO.2,  C1COCCO1.2,  CCC(C)=O.2,  CC#N,  C[N+]([O-])=O.1,  CCCCCCCC.2,  Cc1ccccc1.3,  CCO.3,  C1COCCO1.3,  CCC(C)=O.3,  CC(=O)c1ccccc1,  CC(O)=O.2,  C[N+]([O-])=O.2,  OO,  CCCCCCCC.3,  Cc1ccccc1.4,  CCO.4,  C1COCCO1.4,  CCC(C)=O.4,  CC(O)=O.3,  CCC(O)=O.1,  Nc1ccccc1.1,  C[N+]([O-])=O.3,  OO.1,  CCCCCCCC.4,  Cc1ccccc1.5,  CCO.5,  C1COCCO1.5,  COc1ccccc1,  CCC(C)=O.5,  CCCCN,  C[N+]([O-])=O.4,  CCCCC,  CCCCCC,  CCCCCCCC.5,  C1CCCCC1,  c1ccccc1,  Cc1ccccc1.6,  CO,  CCO.6,  CCCO,  CC(C)O,  CCCCO,  CC(C)(C)O,  CCCCCO,  Oc1ccccc1.1,  CCCCCCO,  Cc1ccccc1O,  Cc1cccc(O)c1.1,  Cc1ccc(O)cc1,  CCCCCCCO,  C1COCCO1.6,  CC(C)=O,  CCC(C)=O.6,  CCCC(C)=O,  CCCCC(C)=O,  CCCCCC(C)=O,  CC(O)=O.4,  CCC(O)=O.2,  CCCC(O)=O.1,  CCCCC(O)=O,  CCCCCC(O)=O,  COC(C)=O,  CCC(=O)OC,  CCOC(C)=O,  CCCOC(C)=O,  CCCCC(=O)OC,  CCCCOC(C)=O,  CCCCCC(=O)OC,  CCCCCOC(C)=O,  CCN,  CNC, ...]
Index: []

[0 rows x 25333 columns]

In [4]:
frames = [features_df, SMILES_df]
result = pd.concat(frames)
display(result)

ABC     ABCGG  nAcid  nBase    SpAbs_A   SpMax_A  SpDiam_A  \
0      4.949747  5.143137    0.0    0.0   9.517541  1.879385   3.75877   
1      5.059137  4.785854    0.0    0.0   8.720566  2.101003  4.202006   
2      1.414214  1.414214    0.0    0.0   2.828427  1.414214  2.828427   
3      4.242641  4.000000    0.0    0.0        8.0       2.0       4.0   
4      3.047207  3.305183    0.0    0.0   5.226252  1.847759  3.695518   
...         ...       ...    ...    ...        ...       ...       ...   
12464  5.766244  5.909452    0.0    1.0   9.910095  2.224503   4.19461   
12465  5.875634  5.943785    1.0    1.0  10.251662  1.961571  3.923141   
12466  5.875634  5.943785    1.0    1.0  10.251662  1.961571  3.923141   
12467  7.289847  7.007240    1.0    0.0  12.784906  1.975377  3.950753   
12468  7.289847  7.007240    1.0    0.0  12.784906  1.975377  3.950753   

          SpAD_A   SpMAD_A   LogEE_A  ...  Cc1c(O)cccc1O.1  \
0       9.517541  1.189693  2.876615  ...              NaN   
1       8.720566  1.245795  2.844305  ...              NaN   
2       2.828427  0.942809  1.849457  ...              NaN   
3            8.0  1.333333  2.687624  ...              NaN   
4       5.226252   1.04525  2.408576  ...              NaN   
...          ...       ...       ...  ...              ...   
12464   9.910095  1.238762  2.984531  ...              NaN   
12465  10.251662  1.139074  3.007497  ...              NaN   
12466  10.251662  1.139074  3.007497  ...              NaN   
12467  12.784906  1.162264  3.210676  ...              NaN   
12468  12.784906  1.162264  3.210676  ...              NaN   

       c1ccc(cc1)c2ccccc2.1  [O-][N+](=O)c1cccc2c1cccc2[N+]([O-])=O  \
0                       NaN                                     NaN   
1                       NaN                                     NaN   
2                       NaN                                     NaN   
3                       NaN                                     NaN   
4                       NaN                                     NaN   
...                     ...                                     ...   
12464                   NaN                                     NaN   
12465                   NaN                                     NaN   
12466                   NaN                                     NaN   
12467                   NaN                                     NaN   
12468                   NaN                                     NaN   

       OC[P]1(=O)Oc2ccccc2c3ccccc13  \
0                               NaN   
1                               NaN   
2                               NaN   
3                               NaN   
4                               NaN   
...                             ...   
12464                           NaN   
12465                           NaN   
12466                           NaN   
12467                           NaN   
12468                           NaN   

       COc1ccc(cc1OC)C(=O)N2CCN(CC2)c3ccc4NC(=O)CCc4c3  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4                                                  NaN   
...                                                ...   
12464                                              NaN   
12465                                              NaN   
12466                                              NaN   
12467                                              NaN   
12468                                              NaN   

       Cc1ccc2[nH]c(Nc3ccccc3)nc2c1   Cc1ccccc1C(O)=O.1  \
0                               NaN                 NaN   
1                               NaN                 NaN   
2                               NaN                 NaN   
3                               NaN                 NaN   
4                               NaN                 NaN   
...  

For tomorrow, does this df look the way it should?
Are the SMILES strings supposed to be indivodual columns? Or is there supposed to be a smiles string column
with the strings as the values inside instead of NA?